patchwork-extensions/src/main/java/com/patchworkmc/impl/extension/PatchworkEntityTypeBuilderExtensions.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.patchworkmc.impl.extension;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+
+/**
+ * Forge does this through patching the constructor instead, we just add methods with mixins instead.
+ */
+public interface PatchworkEntityTypeBuilderExtensions<T extends Entity> {
```

Comment:
One question? why is this class required? You don't need an interface to add methods iirc.

Reply:
Maybe not (i'm not sure from the docs), but not including this makes it impossible to reference our added methods with duck typing.

Reply:
On one hand we don't need this because we don't use this directly, on the other other mods may want to compile against this. I'm still uncertain whether it's necessary.

Reply:
If we have patchwork-patcher patching source code this is required.